list-agreements.lua

```diff
@@ -66,26 +96,49 @@ function is_denied(agr)
     return denied
 end
 
-function generate_output_guild(agr)
-    if is_satisfied(agr) == true then
-        print("Establish a "..get_location_name(agr).." for the "..get_guildhall_profession(agr).." guild, as agreed on "..get_petition_date(agr).." (satisfied)")
-    elseif is_denied(agr) == true then
-        print("Establish a "..get_location_name(agr).." for the "..get_guildhall_profession(agr).." guild, as agreed on "..get_petition_date(agr).." (denied)")
-    else
-        print("Establish a "..get_location_name(agr).." for the "..get_guildhall_profession(agr).." guild, as agreed on "..get_petition_date(agr))
+function is_expired(agr)
+    local expired = false
+    agr_age = get_petition_age(agr)
+    if agr_age[1] ~= 0 then
+        expired = true
     end
+    return expired
 end
 
-function generate_output_temple(agr)
-    if is_satisfied(agr) == true then
-        print("Establish a "..get_location_name(agr).." for "..get_religion_name(agr)..", as agreed on "..get_petition_date(agr).." (satisfied)")
-    elseif is_denied(agr) == true then
-        print("Establish a "..get_location_name(agr).." for "..get_religion_name(agr)..", as agreed on "..get_petition_date(agr).." (denied)")
+--universal handler
+function generate_output(agr,loctype)
+    local loc_name = get_location_name(agr.details[0].data.Location.tier,loctype)
+    local agr_age = get_petition_age(agr)
+    output_str = "Establish a ".. loc_name.." for \""..get_agr_party_name(agr)
+    if loctype == df.abstract_building_type.TEMPLE then
+        output_str = output_str.."\" worshiping '"..get_deity_name(agr)..",'"
+    elseif loctype == df.abstract_building_type.GUILDHALL then
+        output_str = output_str.."\", a "..get_guildhall_profession(agr).." guild,"
     else
-        print("Establish a "..get_location_name(agr).." for "..get_religion_name(agr)..", as agreed on "..get_petition_date(agr))
+        print("Agreement with unknown org")
+        return
+    end
+    output_str = output_str.."\n\tas agreed on "..get_petition_date(agr)..". \t"..agr_age[1].."y, "..agr_age[2].."m, "..agr_age[3].."d ago"
+    if is_satisfied(agr) then
+        print( output_str.." (satisfied)")
+    elseif is_denied(agr) then
+        print(output_str.." (denied)")
+    elseif is_expired(agr) then
+        print(output_str.." (expired)")
+    else
+        print(output_str)
     end
 end
 
+---------------------------------------------------------------------------
+-- Main Script operation
+---------------------------------------------------------------------------
+
+local args = {...}
+local cmd = args[1]
```

Comment:
`cmd` doesn't appear to be used anywhere. however, it would be useful if this script supported the following API:
```
list-agreements
    shows open, unfulfilled agreements
list-agreements all
    shows all agreements, past and present
list-agreements help
    shows script help
```

Reply:
Yup, that was the intention, but i think i neglected to actually do the handler for the arguments - possibly procrastination because i've not done them before 👀

Reply:
The [argparse module](https://docs.dfhack.org/en/stable/docs/Lua%20API.html#argparse) is available if you want to do anything fancy ([example](https://github.com/DFHack/scripts/blob/master/prioritize.lua#L532)), but I think you could get by with an `action_switch` like [this](https://github.com/DFHack/scripts/blob/e4b51365c117f09c3f72fcd16a8ca59ea5c3e5f0/autounsuspend.lua#L29).